modules/CbSystem/main/systemImpl.cpp

```diff
@@ -176,6 +177,56 @@ void systemImpl::check_update_marker() {
     }
 }
 
+void systemImpl::setSystemTime(const std::chrono::time_point<date::utc_clock>& timepoint) {
+    // we ignore:
+    // - enabling or checking of NTP support
+    // - checking the time difference
+    // we do only:
+    // - set the system time with the simplest and most straight-forward method
+    //   (SDBus)
+    // - throw on errors
+    __attribute__((cleanup(sd_bus_error_free))) sd_bus_error sd_error = SD_BUS_ERROR_NULL;
+    __attribute__((cleanup(sd_bus_unrefp))) sd_bus* bus = NULL;
+    const char* bus_timedate_destination = "org.freedesktop.timedate1";
+    const char* bus_timedate_path = "/org/freedesktop/timedate1";
+    const char* bus_timedate_interface = "org.freedesktop.timedate1";
+    constexpr bool relative = false;
+    constexpr bool interactive = false;
+    int rv;
+
+    // date::utc_clock takes leap seconds into consideration, the SDBus system clock does not
+    // so convert to eliminate to 28 seconds difference
+    const auto timepoint_sys_clock = date::clock_cast<std::chrono::system_clock>(timepoint);
+    const auto timepoint_sys_clock_us = std::chrono::time_point_cast<std::chrono::microseconds>(timepoint_sys_clock);
+    const auto usec_utc_sys = timepoint_sys_clock_us.time_since_epoch().count();
+
+    rv = sd_bus_default_system(&bus);
+    if (rv < 0) {
+        throw std::system_error(errno, std::generic_category(), "Could not initialize SDBus");
+    }
+
+    rv = sd_bus_call_method(bus, bus_timedate_destination, bus_timedate_path, bus_timedate_interface, "SetTime",
+                            &sd_error, NULL, "xbb", usec_utc_sys, relative, interactive);
+    if (rv < 0) {
+        throw std::system_error(errno, std::generic_category(), "Could not call SDBus method");
+    }
+
+    rv = sd_bus_flush(bus);
+    if (rv < 0) {
+        throw std::system_error(errno, std::generic_category(), "Could not flush SDBus");
+    }
+
+    sd_bus_close(bus);
```

Comment:
I think this function should close the bus (free resources) under all circumstances (error cases).